bpf/attach.go

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) 2021 Tigera, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package bpf
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path"
+
+	log "github.com/sirupsen/logrus"
+)
+
+type AttachedProgInfo struct {
+	Object string `json:"object"`
+	Hash   string `json:"hash"`
+}
+
+// Check if a hash file exists for an Attach Point and the content matches
+// the Attach Point's fields
+func VerifyProgHash(iface, hook, object string) (bool, string, error) {
+	var (
+		progInfo       AttachedProgInfo
+		bytesToRead    []byte
+		calculatedHash string
+		err            error
+	)
+
+	if calculatedHash, err = sha256OfFile(object); err != nil {
+		return false, "", err
+	}
+
+	name := iface + "_" + hook + ".json"
```

Comment:
Suggest making a utility function to calculate the name of the file; would avoid breaking the link between the two functions